cloudigrade/analyzer/management/commands/analyze_log.py

```diff
@@ -136,3 +170,36 @@ def _is_on_off_event(self, record, valid_events):
             return False
         else:
             return True
+
+    @transaction.atomic
+    def _save_instances_and_events(self, instances):
+        """
+        Save instances and on/off events to the DB.
+
+        Args:
+            instances (dict): Instance data, including a list of events.
+
+        """
+        for instance_id, instance_dict in instances.items():
+            account_id = instance_dict['account_id']
+            # TODO: Don't pull the same account record from the
+            #       DB multiple times. Consider caching.
+            account = Account.objects.get(account_id=account_id)
+            instance, __ = Instance.objects.get_or_create(
+                account=account,
+                ec2_instance_id=instance_id,
+                region=instance_dict['region'],
+            )
+            for event in instance_dict['events']:
+                occurred_at_dt = datetime.datetime.strptime(
+                    event['occurred_at'], '%Y-%m-%dT%H:%M:%SZ'
+                ).replace(tzinfo=tz.tzutc())
+                instance_event = InstanceEvent(
+                    instance=instance,
+                    event_type=event['event_type'],
+                    occurred_at=occurred_at_dt,
+                    subnet=event['subnet'],
+                    ec2_ami_id=event['ec2_ami_id'],
+                    instance_type=event['instance_type'],
+                )
+                instance_event.save()
```

Comment:
You can drop the `instance_event.save()` if you change the previous line from `instance_event = InstanceEvent(` to `InstanceEvent.objects.create(`.